sql/core/src/test/scala/org/apache/spark/sql/StringFunctionsSuite.scala

```diff
@@ -128,6 +128,18 @@ class StringFunctionsSuite extends QueryTest {
     // scalastyle:on
   }
 
+  test("string translate") {
+    val df = Seq(("translate", "")).toDF("a", "b")
+
+    checkAnswer(df.select(translate($"a", "rnlt", "123")), Row("1a2s3ae"))
+    checkAnswer(df.select(translate($"a", "", "123")), Row("translate"))
+    checkAnswer(df.select(translate($"a", "rnlt", "")), Row("asae"))
+    // scalastyle:off
+    // non ascii characters are not allowed in the source code, so we disable the scalastyle.
+    checkAnswer(df.select(translate(lit("花花世界"), "花界", "ab")), Row("aa世b"))
```

Comment:
This is an end-to-end test for DataFrame or SQL parse, these special cases are not necessary here. Instead, we should have one fro SQL parser (using selectExpr())